src/azure-cli/azure/cli/command_modules/search/custom.py

```diff
@@ -17,6 +19,121 @@ def _get_resource_group_location(cli_ctx, resource_group_name):
+        from azure.cli.core.aaz import AAZStrArg
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.ip_rules = AAZStrArg(
+            arg_group="Properties",
+            options=['--ip-rules'],
+            help="Some help",
+            nullable=True,  # allow to remove all the value when it's assigned by null
+        )
+        args_schema.auth_options = AAZStrArg(
+            arg_group="Properties",
+            options=['--auth-options'],
+            help="Some Help",
+            enum=["aadOrApiKey", "apiKeyOnly"],
+        )
+
+        args_schema.ip_rules_internal._registered = False
+        args_schema.api_key_only._registered = False
+        return args_schema
+
+    def pre_operations(self):
+        from azure.cli.core.aaz import has_value
+        import re
+        args = self.ctx.args
+
+        if has_value(args.ip_rules):
+            if args.ip_rules.to_serialized_data() is None or args.ip_rules in [';', ','] :
+                # cleanup all ip_rules
+                args.ip_rules_internal = None
+            else:
+                ip_rules = re.split(';|,', args.ip_rules.to_serialized_data())
+                args.ip_rules_internal = [{"value": ip_rule} for ip_rule in ip_rules]
+
+        if has_value(args.disable_local_auth) and args.disable_local_auth.to_serialized_data() is True:
+            if has_value(args.auth_options):
+                raise MutuallyExclusiveArgumentError("Both the DisableLocalAuth and AuthOptions parameters "
+                                                     "can't be given at the same time")
+            if has_value(args.aad_auth_failure_mode):
+                raise MutuallyExclusiveArgumentError("Both the DisableLocalAuth and AadAuthFailureMode parameters "
+                                                     "can't be given at the same time")
+        if has_value(args.auth_options):
+            if args.auth_options == "apiKeyOnly":
+                if has_value(args.aad_auth_failure_mode):
+                    raise MutuallyExclusiveArgumentError(
+                        "Both an AuthOptions value of apiKeyOnly and an AadAuthFailureMode "
+                        "can't be given at the same time")
+                args.api_key_only = {}
+            elif args.auth_options == "aadOrApiKey" and not has_value(args.aad_auth_failure_mode):
+                raise RequiredArgumentMissingError("An AuthOptions value of aadOrApiKey requires "
+                                                   "an AadAuthFailureMode parameter")
+
+    def pre_instance_update(self, instance):
+        from azure.cli.core.aaz import has_value
+        args = self.ctx.args
+        if has_value(args.auth_options):
+            # clean up current auth_options
+            instance.properties.auth_options = {}
+
+
 # pylint: disable=too-many-locals
 def create_search_service(cmd, resource_group_name, search_service_name, sku, location=None, partition_count=0,
```

Comment:
You can clean up the unused custom function as wee

Reply:
done